### ISSUE TITLE
"late" profiling detail should NOT fall back to `-fprof-auto-top`.

### ISSUE THREAD

**Issue by mpickering** [haskell/cabal#8845]
I think it is potentially very confusing and unexpected that the `late` profiling option falls back to `-fprof-auto-top` if your compiler version is too old. If you are specifying late detail you are not expecting the profiling option to affect the runtime characteristics of your program, because the cost centres should be inserted after profiling.

I think it makes more sense to only place cost centres if your compiler supports the requested mode.

In general, I find it a bit surprising that options are silently ignored if they are not supported, for example, if your compiler doesn't support `parMake` then the `-j` option is ignored without warning.

cc @AndreasPK

**Comment by mpickering** [comment]
I think the correct thing to do is to fail when configuring if you pass `profiling-detail: late` with an unsupported compiler.

This can be achieved by adding a check to `checkCompilerProblems`.

**Comment by AndreasPK** [comment]
Personally I would be more surprised by getting no cost centres with old ghcs than by the fallback which is why I implemented it this way. It also seemed in line with what was done in the past for other options.

If cabal devs feel strongly about changing it I wouldn't them changing it but I have a weak preference for the current implementation.

**Comment by ulysses4ever** [comment]
Indeed, failing would seem to follow the principle of least surprise, and I'd be leaning there. That said, cabal already ignores many possible flag inconsistencies, so I'd not sweat over the current behavior.

**Comment by mpickering** [comment]
See https://discourse.haskell.org/t/debugging-why-late-profiling-detail-still-shows-optics-unexpectedly/6391 for someone in the wild who ran into this. I propose to change it to an error for an unsupported ghc version.

```
profiling-detail = late is only supported by ghc > 9.4
```

**Comment by ulysses4ever** [comment]
We're about to release 3.10.2. would be good to squeeze this in.

**Comment by ulysses4ever** [comment]
Relevant lines are here: https://github.com/haskell/cabal/blob/94c86abdf1ea4d9e79a28b14ca003f381eadf974/Cabal/src/Distribution/Simple/Program/GHC.hs#L679-L681
(the `otherwise` branch). The only question is how to graciously fail at this point. I don't see any calls to `die` in this file. And the code above resides inside a pure function:
```
renderGhcOptions :: Compiler -> Platform -> GhcOptions -> [String]
```
So it feels wrong to `die` in here.
